hack/version.sh

```diff
@@ -105,16 +118,23 @@ function kubevirt::version::ldflags() {
 
     SOURCE_DATE_EPOCH=${KUBEVIRT_SOURCE_DATE_EPOCH-$(git show -s --format=format:%ct HEAD)}
 
+    version_pkg="kubevirt.io/client-go/version"
+
     local buildDate
     [[ -z ${SOURCE_DATE_EPOCH-} ]] || buildDate="--date=@${SOURCE_DATE_EPOCH}"
-    local -a ldflags=($(kubevirt::version::ldflag "buildDate" "$(date ${buildDate} -u +'%Y-%m-%dT%H:%M:%SZ')"))
+    local -a ldflags=($(kubevirt::version::ldflag ${version_pkg} "buildDate" "$(date ${buildDate} -u +'%Y-%m-%dT%H:%M:%SZ')"))
     if [[ -n ${KUBEVIRT_GIT_COMMIT-} ]]; then
-        ldflags+=($(kubevirt::version::ldflag "gitCommit" "${KUBEVIRT_GIT_COMMIT}"))
-        ldflags+=($(kubevirt::version::ldflag "gitTreeState" "${KUBEVIRT_GIT_TREE_STATE}"))
+        ldflags+=($(kubevirt::version::ldflag ${version_pkg} "gitCommit" "${KUBEVIRT_GIT_COMMIT}"))
+        ldflags+=($(kubevirt::version::ldflag ${version_pkg} "gitTreeState" "${KUBEVIRT_GIT_TREE_STATE}"))
     fi
 
     if [[ -n ${KUBEVIRT_GIT_VERSION-} ]]; then
-        ldflags+=($(kubevirt::version::ldflag "gitVersion" "${KUBEVIRT_GIT_VERSION}"))
+        ldflags+=($(kubevirt::version::ldflag ${version_pkg} "gitVersion" "${KUBEVIRT_GIT_VERSION}"))
+    fi
+
+    if [[ -n ${KUBEVIRT_CONTAINER_TAG-} ]]; then
```

Comment:
why there are those `-` at the end all over ?

Reply:
AFAIK specifying `-` will define the variable if it is not defiled and assign empty string to it.
Its being used when the flag `-u` is up and you want to override it.
Not sure why its being used here I guess there is a script that set this flag up, 
I aligned with the style of this script.

Reply:
this file doesnt have -u is it ?
and even if it did, i don't think we should copy the behavior unless we want it (+ verify this is indeed what it meant to in a dummy script, although it sounds logical),
i don't think you need it for KUBEVIRT_CONTAINER_TAG do you ?

Reply:
`hack/version.sh` is sourced by `hack/print-workspace-status.sh`, which includes `set -o nounset` (which is equivalent to `set -u`)

Reply:
We already have a `DOCKER_TAG` which we should probably rename at some point to `KUBEVIRT_CONTAINER_TAG`, but that may be the more accurate tag to use. This one would also default to `devel` for e.g. `make clyster-sync`: https://github.com/kubevirt/kubevirt/blob/main/hack/cluster-build.sh